### PR TITLE
Geneva List VM Size options for master resize

### PR DESCRIPTION
**Pull request by jhoreman**
### Which issue this PR addresses:

Fixes

### What this PR does / why we need it:

This PR supports the ListVMResize Geneva action work on the ARO RP side. 

### Test plan for issue:

<!--
How did you test that this PR works?

We can manually test the functionality of this geneva action call through a local mock geneva actions environment.

### Is there any documentation that needs to be updated for this PR?

Yes, will be written as an SOP for SRE use.

### CODE REVIEW

**Review by bennerv — DISMISSED** [review]
Just mostly naming nits to be more clear about what's happening.  Take it how you wish.  

ResizeOptions rather than Sizes becasue sizes implies sizes for a region, not valid resize options for a VM.

File: pkg/util/azureclient/mgmt/compute/virtualmachines_addons.go [hunk 1]
```diff
@@ -71,3 +72,10 @@ func (c *virtualMachinesClient) List(ctx context.Context, resourceGroupName stri
 
 	return result, nil
 }
+func (c *virtualMachinesClient) ListVMSizes(ctx context.Context, resourceGroupName string, VMName string) (result *[]mgmtcompute.VirtualMachineSize, err error) {
+	future, err := c.VirtualMachinesClient.ListAvailableSizes(ctx, resourceGroupName, VMName)
+	if err != nil {
+		return nil, err
+	}
+	return future.Value, err
```
[the remark this answers is not in the document]
Comment (jhoreman) [reply pkg/util/azureclient/mgmt/compute/virtualmachines_addons.go:80]: Agreed, handled.

File: pkg/util/azureclient/mgmt/compute/virtualmachines_addons.go [hunk 2]
```diff
@@ -16,6 +16,7 @@ type VirtualMachinesClientAddons interface {
 	RedeployAndWait(ctx context.Context, resourceGroupName string, VMName string) error
 	StartAndWait(ctx context.Context, resourceGroupName string, VMName string) error
 	List(ctx context.Context, resourceGroupName string) (result []mgmtcompute.VirtualMachine, err error)
+	ListVMSizes(ctx context.Context, resourceGroupName string, VMName string) (result *[]mgmtcompute.VirtualMachineSize, err error)
```
[the remark this answers is not in the document]
Comment (jhoreman) [reply pkg/util/azureclient/mgmt/compute/virtualmachines_addons.go]: Agreed, handled.

File: pkg/util/azureclient/mgmt/compute/virtualmachines_addons.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bennerv) [reply pkg/util/azureclient/mgmt/compute/virtualmachines_addons.go:80]: looks like it's still using `future.Value`

[the remark this answers is not in the document]
Comment (jhoreman) [reply pkg/util/azureclient/mgmt/compute/virtualmachines_addons.go]: Made update but missed commit, should be good now.

File: pkg/frontend/admin_openshiftcluster_vmsizelist_test.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+						State: api.SubscriptionStateRegistered,
+						Properties: &api.SubscriptionProperties{
+							TenantID: mockTenantID,
+						},
+					},
+				})
+			},
+			mocks: func(tt *test, a *mock_adminactions.MockAzureActions) {
+
+				a.EXPECT().
```
Comment (hawkowl) [inline pkg/frontend/admin_openshiftcluster_vmsizelist_test.go:72]: This test content seems unrelated?

**Review by hawkowl — CHANGES_REQUESTED** [review]
Not sure about the test content -- it seems unrelated to what the Azure API returns?

**Comment by jhoreman** [comment]
Discovered this call does not properly evaluate subscription level usage limits, only region specific limits. Will re-write to look at usage limits based on subscription id + location.

File: pkg/frontend/admin_openshiftcluster_vmsizelist.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	case err != nil:
+		return nil, err
+	}
+
+	subscriptionDoc, err := f.getSubscriptionDocument(ctx, doc.Key)
+	if err != nil {
+		return nil, err
+	}
+
+	a, err := f.azureActionsFactory(log, f.env, doc.OpenShiftCluster, subscriptionDoc)
```
Comment (cadenmarchese) [inline pkg/frontend/admin_openshiftcluster_vmsizelist.go:56]: I'm not sure if this helps with all subscription-specific concerns here, but https://github.com/Azure/ARO-RP/pull/1790 has logic that checks subscription-specific VM availability at install time by using Azure's mgmt compute API (https://github.com/Azure/ARO-RP/pull/1790/files#diff-0bb64ff9a5edb73627cfdbda167ba828c955c1441d9d3eb723c6e06696f1b266R19-R23) using the spAuthorizer.

Do you think it makes sense to move the logic in https://github.com/Azure/ARO-RP/pull/1790 to `pkg/util/computeskus` where we can use it in both places?

**Pull request closed without merging**